src/app/reporting/ReportScheduler.h

```diff
@@ -56,7 +56,7 @@ class ReportScheduler : public ReadHandler::Observer
     class ReadHandlerNode : public IntrusiveListNodeBase<>
     {
     public:
-        using TimerCompleteCallback = void (*)();
+        using TimerCompleteCallback = std::function<void()>;
```

Comment:
It looks like we are basically using this so we can do `[this]() { this->ReportTimerCallback(); }` for the `TimerCompleteCallback`.  That does in fact cause a hidden heap allocation at that point, so it can capture "this"....

But in practice, the subclass ReportTimerCallback is just using `this` for some logging, right?  I wonder whether we could do this in some other way that does not need heap allocation.  I guess most simply we could store a pointer to the scheduler in the nodes instead of in the callback, and either use pointer-to-member for the callback type or use a bare function that gets handed the scheduler and then have a virtual getter for the function pointer to use for that....

Reply:
My mistake, it seems something was missing in my last commit, I will update it shortly

Reply:
Now passing a pointer to the report scheduler instead and using `mScheduller->ReportTimerCallback();`